torch/jit/__init__.py

```diff
@@ -1731,6 +1731,31 @@ def __getstate__(self):
                 "Mixed serialization of script and non-script modules is not supported. " +
                 "For purely script modules use my_script_module.save(<filename>) instead.")
 
+        # Python magic methods do lookups on an object's class type, instead of looking up
+        # the desugared attribute. In order to continue to expose the magic methods
+        # of builtin-containers (ModuleList, Sequential, ModuleDict) to python we
+        # define magic methods here as a shim to the correct attribute.
+        def forward_magic_method(self, method_name, *args, **kwargs):
+            self_method = getattr(self, method_name)
+            if self_method == getattr(RecursiveScriptModule, method_name):
+                raise NotImplementedError()
+            return self_method(*args, **kwargs)
+
+        def __iter__(self):
+            return self.forward_magic_method("__iter__")
+
+        def __getitem__(self, idx):
```

Comment:
Defining the methods this way instead of something like `_ConstModuleList` before will make the error messages confusing for something like re-assigning a module since it would say it has no `__setitem__` rather than saying something about how compiled `ModuleList`s are constant

Reply:
Could you give an example of when you think the error messages would be confusing ? I included an error message above when this code path gets called: "object is not iterable"